spanner/json_column_type_samples.rb

```diff
@@ -0,0 +1,137 @@
+  # [START spanner_update_data_with_json_column]
+  # project_id  = "Your Google Cloud project ID"
+  # instance_id = "Your Spanner instance ID"
+  # database_id = "Your Spanner database ID"
+
+  require "google/cloud/spanner"
+
+  spanner = Google::Cloud::Spanner.new project: project_id
+  client  = spanner.client instance_id, database_id
+
+  rows = [{
+    VenueId: 1,
+    VenueDetails: { rating: 9, open: true }
+  }]
+  client.update "Venues", rows
+
+  # VenueDetails must be specified as a string, as it contains a top-level
+  # array of objects that should be inserted into a JSON column. If we were
+  # to specify this value as an array instead of a string, the client
+  # library would encode this value as ARRAY<JSON> instead of JSON.
+  rows = [{
+    VenueId: 2,
+    VenueDetails: [
+      {
+        name: "room 1",
+        open: true
+      },
+      {
+        name: "room 2",
+        open: false
+      }
+    ].to_json
+  }]
+  client.update "Venues", rows
+
+  puts "Rows are updated."
+  # [END spanner_update_data_with_json_column]
+end
+
+def query_with_json_params project_id:, instance_id:, database_id:
+  # [START spanner_query_with_json_parameter]
+  # project_id  = "Your Google Cloud project ID"
+  # instance_id = "Your Spanner instance ID"
+  # database_id = "Your Spanner database ID"
+
+  require "google/cloud/spanner"
+
+  spanner = Google::Cloud::Spanner.new project: project_id
+  client = spanner.client instance_id, database_id
+
+  query = "SELECT VenueId, VenueDetails FROM Venues
+    WHERE JSON_VALUE(VenueDetails, '$.rating') = JSON_VALUE(@details, '$.rating')"
+  result = client.execute_query query,
+                                params: { details: { rating: 9 } },
+                                types: { details: :JSON }
+
+  result.rows.each do |row|
+    puts "VenueId: #{row['VenueId']}, VenueDetails: #{row['VenueDetails']}"
+  end
+  # [END spanner_query_with_json_parameter]
+end
+
+def usage
+  puts <<~USAGE
+    Usage: bundle exec ruby json_column_type_samples.rb [command] [arguments]
+
+    Commands:
+      add_json_column               <instance_id> <database_id> Add JSON column to table.
+      update_data_with_json_column  <instance_id> <database_id> Updates rows JSON field value.
+      query_with_json_params        <instance_id> <database_id> Query data using JSON datatype.
+
+    Environment variables:
+      GOOGLE_CLOUD_PROJECT must be set to your Google Cloud project ID
+  USAGE
+end
+
+def run_sample arguments
+  commands = [
+    "add_json_column", "update_data_with_json_column", "query_with_json_params"
+  ]
+
+  command = arguments.shift
+  project_id = ENV["GOOGLE_CLOUD_PROJECT"]
+
+  return usage unless commands.include? command
+
+  sample_method = method command
+  parameters = { project_id: project_id }
+
+  sample_method.parameters.each do |paramater|
+    next if paramater.last == :project_id
+    parameters[paramater.last] = arguments.shift
+  end
+
+  sample_method.call(**parameters)
```

Comment:
Can we simplify this to: 

```
  sample_method.call(project_id, *arguments)
```

I am not sure if this would work.

Reply:
Updated.